File: lab4 mpp/TeledonPersistence/CazDBRepository.cs
using teledon.domain;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace teledon.repository
{
    public class CazDBRepository : ICazRepo
    {
        private static readonly ILog log = LogManager.GetLogger("RepoCaz");

        public CazCaritabil Delete(long id)
        {
            log.InfoFormat("entering delete cazuri with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Cazuri where idCaz=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();

            }
            log.InfoFormat("exiting delete cazuri");
            return null;
        }

        public IEnumerable<CazCaritabil> FindAll()
        {
            log.InfoFormat("Entering findAll Cazuri");

            IDbConnection con = DBUtils.getConnection();
            IList<CazCaritabil> cazuri = new List<CazCaritabil>();
            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Cazuri";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        String Descriere = dataR.GetString(1);
                        int sumaAdunata = dataR.GetInt32(2);
                        CazCaritabil caz = new CazCaritabil(Descriere, sumaAdunata);
                        caz.ID = id;

                        //Console.WriteLine(voluntar);
                        cazuri.Add(caz);
                    }
                }
            }

            log.InfoFormat("Exiting findAll cazuri with the result from it");
            return cazuri;
        }

        public IEnumerable<Donator> FindAllDonatori(CazCaritabil caz)
        {
            throw new NotImplementedException();
        }

        public CazCaritabil findByName(string nume)
        {
            log.InfoFormat("Entering findByNume caz with value {0}", nume);
            IDbConnection con = DBUtils.getConnection();

            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Cazuri where descriere=@id";

                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = nume;
                comm.Parameters.Add(paramId);


                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idCaz = dataR.GetInt32(0);
                        String Descriere = dataR.GetString(1);
                        int sumaAdunata = dataR.GetInt32(2);
                        CazCaritabil caz = new CazCaritabil(Descriere, sumaAdunata);
                        caz.ID = idCaz;
                        log.InfoFormat("Exiting findOne with value {0}", caz);
                        return caz;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public CazCaritabil FindOne(long id)
        {
            log.InfoFormat("Entering findOne caz with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Cazuri where idCaz=@id";

                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);


                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idCaz = dataR.GetInt32(0);
                        String Descriere = dataR.GetString(1);
                        int sumaAdunata = dataR.GetInt32(2);
                        CazCaritabil caz = new CazCaritabil(Descriere, sumaAdunata);
                        caz.ID = idCaz;
                        log.InfoFormat("Exiting findOne with value {0}", caz);
                        return caz;
                    }
                }
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public CazCaritabil Save(CazCaritabil entity)
        {
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "Insert into Cazuri(descriere, sumaAdunata) values (@descriere,@sumaAdunata)";
                IDbDataParameter paramDescriere = comm.CreateParameter();
                paramDescriere.ParameterName = "@descriere";
                paramDescriere.Value = entity.Nume;
                comm.Parameters.Add(paramDescriere);

                IDbDataParameter paramSuma = comm.CreateParameter();
                paramSuma.ParameterName = "@sumaAdunata";
                paramSuma.Value = entity.SumaAdunata;
                comm.Parameters.Add(paramSuma);

                var result = comm.ExecuteNonQuery();

                if (result == 0)
                    log.InfoFormat("save exited with value {0}", result);
                else
                    log.InfoFormat("save exited with value {0}", result);

            }


            return null;
        }



        public CazCaritabil Update(CazCaritabil entity)
        {
            log.InfoFormat("Entering updateCaz");
            IDbConnection con = DBUtils.getConnection();

            Console.WriteLine(con);
            long id = entity.ID;
            if(FindOne(id) == null)
            {
                throw new ArgumentException("This id does not exist..");
            }

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "UPDATE Cazuri SET sumaAdunata = @suma where idCaz = @id";
                IDbDataParameter paramSuma = comm.CreateParameter();
                paramSuma.ParameterName = "@suma";
                paramSuma.Value = entity.SumaAdunata;
                comm.Parameters.Add(paramSuma);

                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                var result = comm.ExecuteNonQuery();
                log.InfoFormat("Update {0} values", result);

            }
            log.InfoFormat("Updated {0} ", entity);
            return null;
        }
    }
}

File: GUImpp/TeledonPersistence/src/main/java/teledon/persistence/repository/CazDBRepository.java
package teledon.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import teledon.model.CazCaritabil;
import teledon.model.Donator;

import java.io.FileInputStream;
import java.io.IOException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;
import org.springframework.stereotype.Component;

@Component
public class CazDBRepository implements ICazRepo {
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();

    public CazDBRepository(Properties properties) {
        logger.info("Initializing CazDBRepository with properties: {} ", properties);
        //dbUtils = new JdbcUtils(properties);
        try {
            properties.load(new FileInputStream("C:\\Users\\Patras Sergiu\\Desktop\\GUImpp\\l3\\TeledonServer\\src\\main\\resources\\teledonServer.properties"));
        } catch (IOException e) {
            e.printStackTrace();
        }
        dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Iterable<Donator> findAllDonatori(CazCaritabil caz) {
        return null;
    }

    @Override
    public CazCaritabil findOne(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        CazCaritabil caz = new CazCaritabil();
        try(PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Cazuri WHERE idCaz=?")) {
            preparedStatement.setLong(1, aLong);
            ResultSet result = preparedStatement.executeQuery();
            int id = result.getInt("idCaz");
            String descriere = result.getString("descriere");
            int suma = result.getInt("sumaAdunata");
            caz = new CazCaritabil(descriere, suma);
            caz.setId(aLong);

        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " +e);
        }
        logger.traceExit(caz);
        return caz;
    }

    @Override
    public Iterable<CazCaritabil> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<CazCaritabil> cazuri = new ArrayList<>();

        try (PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Cazuri")) {
            try(ResultSet result = preparedStatement.executeQuery()) {
                System.out.println("prepare");
                while(result.next()) {
                    System.out.println("result");
                    Long id = result.getLong("idCaz");
                    String descriere = result.getString("descriere");
                    int sumaAdunata = result.getInt("sumaAdunata");
                    CazCaritabil caz = new CazCaritabil(descriere, sumaAdunata);
                    caz.setId(id);
                    System.out.println(caz);
                    cazuri.add(caz);
                }
            }
        } catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(cazuri);
        return cazuri;
    }

    @Override
    public CazCaritabil save(CazCaritabil entity) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into Cazuri(descriere, sumaAdunata) values (?,?)")){
            preparedStatement.setString(1, entity.getNume());
            preparedStatement.setInt(2, entity.getSumaAdunata());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }
        catch(SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public CazCaritabil delete(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("DELETE FROM Cazuri WHERE idCaz = ?")){
            preparedStatement.setInt(1, aLong.intValue());
            int result = preparedStatement.executeUpdate();
            logger.trace("Deleted {} instances", result);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        return null;
    }

    @Override
    public CazCaritabil update(CazCaritabil entity) {
        System.out.println("Repo " + entity);
            logger.traceEntry();
            Connection con = dbUtils.getConnection();
            Long id = entity.getId();
            if(findOne(id) == null) {
                throw new IllegalArgumentException("This id doesn't exist..");
            }

            String query = "UPDATE Cazuri SET sumaAdunata = ? WHERE idCaz = ?";
            try(PreparedStatement preparedStatement = con.prepareStatement(query)){
                preparedStatement.setInt(1, entity.getSumaAdunata());
                preparedStatement.setLong(2, entity.getId());
                int result = preparedStatement.executeUpdate();
                logger.trace("Updated {} instances", result);
            }
            catch(SQLException e) {
                logger.error(e);
                System.err.println("Error DB " +  e);
            }

        return findOne(id);
    }
}
